src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -295,7 +302,27 @@ private void perform(Run<?,?> build, Launcher launcher, TaskListener listener) t
     	final IProgress progress = new ScanProgress(listener);
     	final boolean suspend = m_wait;
     	final IScan scan = ScanFactory.createScan(getScanProperties(build, listener), progress, m_authProvider);
-        
+		JenkinsAuthenticationProvider checkAppScan360Connection = new JenkinsAuthenticationProvider(m_credentials,build.getParent().getParent());
+        if(m_type.equals("Dynamic Analyzer") && checkAppScan360Connection.isAppScan360()){
```

Comment:
Rather than having all of these checks, can we just disable the options for AppScan 360 credentials? It would be much cleaner that way instead of having to check all of these different combinations of service urls and option values.